src/app/mesh/qgsmeshrenderervectorsettingswidget.h

```diff
@@ -69,6 +90,12 @@ class APP_EXPORT QgsMeshRendererVectorSettingsWidget : public QWidget, private U
 
     QgsMeshLayer *mMeshLayer = nullptr; //not owned
     int mActiveDatasetGroup = -1;
+
+    QList<QWidget *> mArrowSettingsWidget;
```

Comment:
I would group the arrow settings widget in one new parent widget in ui file. same with streamlines widgets. and then just hide the whole group.